app/controllers/application_controller/ci_processing.rb

```diff
@@ -250,11 +250,13 @@ def cloud_object_store_button_operation(klass, task)
   def get_rec_cls
     # FIXME: the specs for ci_processing rely on setting (and testing) request.parameters['controller'].
     # That is wrong and needs to be fixed.
+    return VmOrTemplate if display_vm_template?
```

Comment:
This is red flag for me, there's a big conditional statement below and you're bypassing it and so increasing the code complexity.